### PR TITLE
Add page for related project/tools

### LINKED ISSUE #1621: [Suggestion / New Feature?] Use of Built-in Method to Visualize the File Hierarchy
Hello! 

I'm new to h5py and I don't know if this can cause any issues with the current implementation.

The idea of this suggestion is to add a built-in method capable of showing the file structure in a kind of 'file system tree' . I'm aware that there are ways to visualize it, such as [punx](https://punx.readthedocs.io/en/latest/source_code/h5tree.html), but I wonder if this visualization could be performed without the need of downloading and managing additional dedicated packages.

With that in mind, my suggestion would be something like:

Code:
```python
import h5py

with h5py.File('some/hf5/file.h5', 'r') as hdf_file:
    hdf_file.tree_view() # or 'file_structure_view()', 'structure_view()' or simply 'view()'...
```

Resulting in:
```
HDF5 File at /path/to/hdf5/file.h5

/
    dataset_0
    Group1
        dataset_1
        dataset_2
    Group2
        Group_3
            dataset_3
        dataset_4
```

Some possible parameters could be:

 `verbose`: specifying how much information is displayed in each node of the tree (e.g. show only keys, show keys and first elements of a dataset, show keys and attributes...);
 `file_output`: file to which the output can be redirected;
`indentation`: number of tabs to be used in file structure formatting

 * Operating System: Ubuntu 20.04 LTS
 * Python version: 3.7.7
 * Where Python was acquired: Mini Conda
```
python -c 'import h5py; print(h5py.version.info)'

h5py    2.10.0
HDF5    1.10.4
Python  3.7.7 (default, May  7 2020, 21:25:33) 
[GCC 7.3.0]
sys.platform    linux
sys.maxsize     9223372036854775807
numpy   1.19.1
```

### PR DESCRIPTION
**Pull request by aragilar**
This closes #1621 in that we would have a page for listing other projects users may be interested in. I've also moved the ipython extension docs onto this page, as they were quite hidden otherwise on the config page.

Definitely not complete, people should feel free to push links to projects.

### CODE REVIEW

**Comment by scopatz** [comment]
CC @asmeurer - please add versioned-hdf5 to this page

File: docs/related_projects.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+   def main():
+       ip.ex('from h5py import ipy_completer')
+       ip.ex('ipy_completer.load_ipython_extension()')
+
+Exploring and Visualising HDF5 files
+------------------------------------
+
+h5py does not contain a tool for exploring or visualising HDF5 files, but tools
+built on h5py that can display the structure of h5py include:
```
[the remark this answers is not in the document]
Comment (takluyver) [reply docs/related_projects.rst:64]: The project I'm part of also has [a list of viewer tools](https://github.com/panosc-eu/panosc/blob/master/Work%20Packages/WP4%20Data%20analysis%20services/resources/hdf5-viewers.rst). But there are quite a few, and I'm not very good at keeping up with what's still developed and supported.

File: docs/related_projects.rst [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+
+Field/Format Specific Libraries
+-------------------------------
+
+The are projects which build upon h5py to read and write to HDF5 files with a
+specific schema, users of these formats may be interested in using these
+higher-level wrappers of h5py:
+
+ *
```
[the remark this answers is not in the document]
Comment (aragilar) [reply docs/related_projects.rst:87]: Sure, let's drop it. I don't know of any astronomy standards for HDF5 (radio astronomy is apparently experimenting with HDF5, but most people are sticking to FITS). I figured putting this here would provoke lots of suggestions for libraries, but if there are no obvious ones, then there's not much point having it.

File: docs/related_projects.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (aragilar) [reply docs/related_projects.rst]: I've added those links.

File: docs/related_projects.rst [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+It is also possible to configure IPython to enable the completer every time you
+start a new session. For >=ipython-0.11, "h5py.ipy_completer" just needs to be
+added to the list of extensions in your ipython config file, for example
+:file:`~/.config/ipython/profile_default/ipython_config.py` (if this file does
+not exist, you can create it by invoking `ipython profile create`)::
+
+   c = get_config()
+   c.InteractiveShellApp.extensions = ['h5py.ipy_completer']
+
+For <ipython-0.11, the completer can be enabled by adding the following lines
```
[the remark this answers is not in the document]
Comment (aragilar) [reply docs/related_projects.rst]: Done

**Comment by aragilar** [comment]
I've added pytables to the intro, with links to the two faq entries.

Some other projects we may want to link to:
https://github.com/DaanVanVugt/h5pickle - I think I've seen this mentioned in issues
https://github.com/telegraphic/hickle - same (I'm unclear exactly how things are stored).
https://github.com/h5preserve/h5preserve - the wrapper I wrote (does versioning and uses as similar philosophy to http://eev.ee/blog/2015/10/15/dont-use-pickle-use-camel/)

**Comment by takluyver** [comment]
I think all of those could be on the page, but I'm struggling to see how we categorise things and make it clear what they all are.

On a related note: seeing h5preserve for the first time, it took a surprising amount of reading to work out what it was. The README describe it as "a thin wrapper around h5py" - but what does it make more convenient? How do you use it? After reading the blog post about Camel I can kind of guess the rough shape of it. The docs landing page provides some explanation, but it's still not exactly clear what it's offering. The quickstart page finally explains it and demonstrates it, but it's not very amenable to scanning - you really need to read through it to make sense of it.

Anyway, feel free to take all that with a big grain of salt. Sometimes it can be useful to know what a project looks like to someone seeing it for the first time, but I'm just one person and it's totally possible that I missed something that's generally obvious.

**Comment by aragilar** [comment]
The page was initially for viewers and external filters (I thought there were other projects providing filters, as well as hdf5plugin), but I figured there were a bunch of projects building on h5py, it made sense to try to collect them together (rather than potentially having them scattered throughout the docs). As is, I think it's fine (and if someone wants to merge this, feel free to do it), but I figured there were widely used wrappers for specific formats, which may be better for users that using raw h5py (and potentially missing out key metadata).

With respect to h5preserve, that's definitely useful feedback, I still haven't really clarified how to think about it independent of camel, it does its job and I can think of it as a black box most of the time (it was initially a day or so of hacking, after repeatedly getting annoyed that I needed to worry about handling older versions of output in the middle of science code). I should probably sit down and refresh the docs at some point.

**Comment by asmeurer** [comment]
Here is a blurb for versioned-hdf5. It isn't clear to me where this would go on the current page:

Versioned-hdf5 (https://deshaw.github.io/versioned-hdf5/) is a library that builds a versioned abstraction on top of h5py. It provides a wrappers around the h5py API that allows storing different versions of groups and datasets within an HDF5 file.

**Review by tacaswell — APPROVED** [review]

**Pull request merged**